pkg/applier/manager.go

```diff
@@ -148,44 +149,53 @@ func (m *Manager) runWatchers(ctx context.Context) error {
 	}
 }
 
-func (m *Manager) createStack(ctx context.Context, name string) error {
+func (m *Manager) createStack(ctx context.Context, name string) {
 	// safeguard in case the fswatcher would trigger an event for an already existing stack
 	if _, ok := m.stacks[name]; ok {
-		return nil
-	}
-	m.log.WithField("stack", name).Info("registering new stack")
-	sa, err := NewStackApplier(ctx, name, m.KubeClientFactory)
-	if err != nil {
-		return err
+		return
 	}
 
+	stackCtx, cancelStack := context.WithCancel(ctx)
+	stack := stack{cancelStack, NewStackApplier(name, m.KubeClientFactory)}
+	m.stacks[name] = stack
+
 	go func() {
-		_ = sa.Start()
-	}()
+		log := m.log.WithField("stack", name)
+		for timer := time.NewTimer(0 * time.Second); ; timer.Reset(10 * time.Second) {
```

Comment:
What is the idea behind this? Maybe `time.Ticker` would be more readable and straightforward.

Reply:
This code should run the stack, as long as the context is not yet done. The stack's `Run` method will only return if either the context is done, or if an error occurs. In the latter case, the stack should be re-run after some timeout (10 seconds in this case).

I think a more readable version of the above would be this:

```go

go func() {
	log := m.log.WithField("stack", name)
	for {
		log.Info("Running stack")
		if err := stack.Run(stackCtx); err != nil {
			log.WithError(err).Error("Failed to run stack")
		}

		select {
		case <-time.After(10 * time.Second):
			continue
		case <-stackCtx.Done():
			log.Info("Stack done")
			return
		}
	}
}()
```

Reply:
Done.